src/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable.js

```diff
@@ -65,31 +53,45 @@ export class VisualizeEmbeddable extends Embeddable  {
         this.uiState.set(key, customization[key]);
       });
       this.customization = customization;
-      isDirty = true;
       this.uiState.on('change', this._uiStateChangeHandler);
     }
+  }
 
+  /**
+   * Retrieve the panel title for this panel from the container state.
+   * This will either return the overwritten panel title or the visualization title.
+   */
+  getPanelTitle(containerState) {
     let derivedPanelTitle = '';
     if (!containerState.hidePanelTitles) {
       derivedPanelTitle = containerState.customTitle !== undefined ?
         containerState.customTitle :
         this.savedVisualization.title;
     }
+    return derivedPanelTitle;
+  }
+
+  onContainerStateChanged(containerState) {
+    this.transferCustomizationsToUiState(containerState);
+
+    const updatedParams = {};
+
+    // Check if timerange has changed
+    if (containerState.timeRange !== this.timeRange) {
+      updatedParams.timeRange = containerState.timeRange;
+      this.timeRange = containerState.timeRange;
+    }
 
+    const derivedPanelTitle = this.getPanelTitle(containerState);
     if (this.panelTitle !== derivedPanelTitle) {
+      updatedParams.dataAttrs = {
+        title: derivedPanelTitle,
+      };
       this.panelTitle = derivedPanelTitle;
-      isDirty = true;
     }
 
-    if (isDirty && this.handler && this.domNode) {
-      // TODO: We need something like this in the handler
-      // this.handler.update(this.getHandlerParams());
-      // For now:
-      this.destroy();
-      this.handler = this.loader.embedVisualizationWithSavedObject(
-        this.domNode,
-        this.savedVisualization,
-        this.getHandlerParams());
+    if (this.handler && Object.keys(updatedParams).length > 0) {
```

Comment:
Possibly better expression of intent: `if (this.handler && !_.isEmpty(updatedParams)) {`

Reply:
Why use lodash if you can use vanilla JS :D But yeah I agree it's more semantic, so I will change it.

Reply:
Ha. Well, I totally agree with that philosophy (using vanilla where possible)... I just wish vanilla JS had a better standard lib!